### DISCUSSION TITLE
Setting annotation within CRD

### DISCUSSION THREAD

**Discussion by agentp-1337** [arkmq-org/arkmq-org-broker-operator#437] [General] [answered]
Within the deploymentPlan of the ActiveMQArtemis CRD an annotation part is needed.
Reason: When using istio, we have to set the following annotation within the STS so the used ports (7801 - 7810) can communicate outside of istio:
- traffic.sidecar.istio.io/excludeInboundPorts
- traffic.sidecar.istio.io/excludeOutboundPorts
(https://istio.io/latest/docs/reference/config/annotations/)

If we cannot set these annotation istio will block the communication on those ports due to the fact that no service definition are used (headless) for them.

Thx considering to add the option to define annotation.

P.S. Please check if this annoations would be needed within the drain-job or not, thx.

**Comment by brusdev** [comment, marked as answer]
@agentp-1337 adding a property to customize broker pod annotations makes sense to me. Feel to create an issue and eventually a pull request, of course community contributions are always welcome :-)

**Reply by brusdev** [reply]
@gaohoward has just created both an issue and the related request to set annotation within CRD, see https://github.com/artemiscloud/activemq-artemis-operator/pull/451

**Comment by lordigon** [comment]
@agentp-1337 I would like to setup istio in Artemis but I have some issues can you please tell me if you used Istio PeerAuthentication resource along with traffic.sidecar.istio.io/excludeInboundPorts. I would like to exclude Jolokia tcp port but it doesn't work.
Did you enabled  istio also with operator POD?
